sdk/servicebus/azure-servicebus/README.md

```diff
@@ -227,20 +357,47 @@ special "dead-letter sub-queue" where it can be accessed using the `get_queue_de
 
 There are various timeouts a user should be aware of within the library.
 - 10 minute service side link closure:  A link, once opened, will be closed after 10 minutes idle to protect the service against resource leakage.  This should largely
-be transparent to a user, but if you notice a reconnect occuring after such a duration, this is why.  Performing any operations, including management operations, on the
+be transparent to a user, but if you notice a reconnect occurring after such a duration, this is why.  Performing any operations, including management operations, on the
 link will extend this timeout.
 - idle_timeout: Provided on creation of a receiver, the time after which the underlying UAMQP link will be closed after no traffic.  This primarily dictates the length
 a generator-style receive will run for before exiting if there are no messages.  Passing None (default) will wait forever, up until the 10 minute threshold if no other action is taken.
-- max_wait_time: Provided when calling receive() to fetch a batch of messages.  Dictates how long the receive() will wait for more messages before returning, similarly up to the aformentioned limits.
+- max_wait_time: Provided when calling receive() to fetch a list of messages.  Dictates how long the receive() will wait for more messages before returning, similarly up to the aformentioned limits.
 
-### AutoLockRenew
+> **NOTE:** If processing of a message or session is sufficiently long as to cause timeouts, as an alternative to calling `renew_lock()` manually, one can
+> leverage the `AutoLockRenew` functionality detailed below.
+
+### [AutoLockRenew][autolockrenew_reference]
+
+`AutoLockRenew` is a simple method for ensuring your message or session remains locked even over long periods of time, if calling `renew_lock()` is impractical or undesired.
+Internally, it is not much more than shorthand for creating a concurrent watchdog to call `renew_lock()` if the object is nearing expiry.
```

Comment:
Hahahaha you're not _WRONG_ but I was being more vague than that slightly on purpose:  The user doesn't need to know specifically how it's doing it here other than that there is a concurrent thread doing a keepalive.

Let me know if you think the way I wrote it was deceptive, I intentionally "left out implementation details" to phrase it as a common concept (parallel watchdog) but don't want to mislead.

Reply:
On further thought I kinda wouldn't even want to introduce that at this point, since for most cases the user wouldn't want or need to tweak that (similar to max worker count).  As always don't hesitate to shout if you disagree.

Reply:
that's true, most users doesn't need to know how we implement this. Let's revisit this in the future if there's any user feedback.

I'm good with the current statement.